lib/storage/metadata/MetadataWrapper.js

```diff
@@ -515,6 +517,75 @@ class MetadataWrapper {
                 return cb();
             });
     }
+
+    /**
+     * Put bucket indexes
+     *
+     * indexSpec format:
+     * [
+     *      { key: Map<key, value>, name: <id 1>, ... , < backend options> },
+     *      ...
+     *      { key: Map<key, value>, name: <id n>, ... },
+     * ]
+     *
+     *
+     * @param {String} bucketName bucket name
+     * @param {Array<Object>} indexSpecs index specification
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
+    putBucketIndexes(bucketName, indexSpecs, log, cb) {
+        log.debug('put bucket indexes');
+
+        if (!supportsIndexingMDBackends.has(this.implName)) {
```

Comment:
Maybe a more straightforward way to check if the backend supports it is checking if the client object has the `putBucketIndexes` function